crates/flox/src/commands/general.rs

```diff
@@ -132,9 +146,48 @@ pub enum GeneralCommands {
     ResetMetrics,
 
     /// access to the nix CLI
+    #[bpaf(command)]
     Nix(#[bpaf(external(parse_nix_passthru))] WrappedNix),
 }
 
+#[allow(unused)] // pending rust implementation
+#[derive(Bpaf, Clone)]
+pub struct LoginArgs {
+    /// the hostname of the flox hub to authenticate with
+    #[bpaf(long, short, argument("hostname"))]
+    hostname: String,
+    /// save authentication credentials in plain text instead of credential store
+    #[bpaf(long)]
+    insecure_storage: bool,
+    /// read token from standard input
+    #[bpaf(long)]
+    with_token: bool,
+    /// Show login status
+    status: bool,
+}
+
+/// handling of login, pending rust implementation
+impl LoginArgs {
+    async fn handle(&self, flox: Flox) -> Result<()> {
+        flox_forward(&flox).await
+    }
+}
+
+#[allow(unused)] // pending rust implementation
+#[derive(Bpaf, Clone)]
+pub struct LogoutArgs {
+    /// the hostname of the flox hub to log out of
+    #[bpaf(long, short, argument("hostname"))]
+    hostname: String,
```

Comment:
I guess that is optional, right?

```suggestion
    hostname: Option<String>,
```